packages/atomic/src/components/insight/user-actions/atomic-insight-user-actions-timeline/atomic-insight-user-actions-timeline.tsx

```diff
@@ -0,0 +1,242 @@
+import {Component, h, State, Prop} from '@stencil/core';
+import {
+  buildInsightUserActions,
+  InsightUserActions,
+  InsightUserActionsState,
+  InsightUserSession,
+} from '../..';
+import ArrowDownIcon from '../../../../images/arrow-full-down.svg';
+import ArrowUpIcon from '../../../../images/arrow-up.svg';
+import {
+  InitializableComponent,
+  InitializeBindings,
+  BindStateToController,
+} from '../../../../utils/initialization-utils';
+import {Button} from '../../../common/button';
+import {NoItemsContainer} from '../../../common/no-items/container';
+import {MagnifyingGlass} from '../../../common/no-items/magnifying-glass';
+import {InsightBindings} from '../../atomic-insight-interface/atomic-insight-interface';
+
+/**
+ * This component displays all the actions performed by a user around the time they created a case.
+ * The actions are grouped into multiple sessions, including the session during which the case was created,
+ * the sessions preceding the case creation and the sessions following the case creation.
+ *
+ * @component
+ * @example
+ * <AtomicInsightUserActionsTimeline userId={'123'} caseCreationDate={'2024-08-15T10:00:00Z'} />
+ *
+ */
+@Component({
+  tag: 'atomic-insight-user-actions-timeline',
+  styleUrl: 'atomic-insight-user-actions-timeline.pcss',
+  shadow: true,
+})
+export class AtomicInsightUserActionsTimeline
+  implements InitializableComponent<InsightBindings>
+{
+  @InitializeBindings() public bindings!: InsightBindings;
+  public userActions!: InsightUserActions;
+
+  @BindStateToController('userActions')
+  @State()
+  public userActionsState!: InsightUserActionsState;
+  public error!: Error;
+
+  /**
+   * The ID of the user whose actions are being displayed.
+   */
+  @Prop() public userId!: string;
+  /**
+   * The date and time when the case was created.
```

Comment:
I would also give an example here, so people reading the reference doc have an example of the expected formatting of the data they're supposed to set.

Reply:
Also there could be an argument made so the client-facing param is called `dateTime` instead of just `date` since we're looking for more than just a date? (I don't know the answer.)

Reply:
Indeed, the usual use case is that a date and time would be passed here as an ISO date, so this makes sense!
I will update the name of the property for more precision.